### ISSUE TITLE
What does WCAG require of SVGs to pass SC 1.1.1

### ISSUE THREAD

**Issue by mraccess77** [w3c/wcag#840]
Might be worth discussing what is required to pass SC 1.1.1 with an SVG.
1. Does an SVG that does not have role img need to be marked as decorative to pass?  From what I can tell it depend on child elements like title, text, etc.?  Are roles presentation and none suitable in all situations? or is aria-hidden required when certain child nodes are present -- role presentation not allowed when title element present.  See link below that provides useful discussion but is not complete in IMO.
2. What methods provide an accessible name for an SVG?  Text? title, aria-label, etc.  

Of use : https://www.w3.org/TR/svg-aam-1.0/#include_elements

**Comment by jake-abma** [comment]
Hi @mraccess77 

> Does an SVG that does not have role img need to be marked as decorative to pass?

Decoration seems to be covered by a must to implement it in a way it can be ignored:

> Decoration, Formatting, Invisible: If non-text content is pure decoration, is used only for visual formatting, or is not presented to users, then it is implemented in a way that it can be ignored by assistive technology.

If not decoration it is non-text content and needs a text alternative that serves the equivalent purpose. Role "img" is not demanded by the SC.

roles presentation and none, aria-hidden etc. are not mentioned by the SC but option to get to the result. Up to the content author to make right decisions.

> What methods provide an accessible name for an SVG? Text? title, aria-label, etc.

You probably already know but lots of different options available and not all well supported for SVG, see the blog from Scott, but there are more: https://www.scottohara.me/blog/2019/05/22/contextual-images-svgs-and-a11y.html

But specific techniques are not required for the SC.

If we add SVG (like ASCII Art) to the NOTE of the definition of "non-text content" will that make it clear?

any content that is not a sequence of characters that can be programmatically determined or where the sequence is not expressing something in human language

> **non-text content**
> any content that is not a sequence of characters that can be programmatically determined or where the sequence is not expressing something in human language
> 
> Note: This includes **SVG**, ASCII Art (which is a pattern of characters), emoticons, leetspeak (which uses character substitution), and images representing text

https://www.w3.org/TR/UNDERSTANDING-WCAG20/text-equiv-all.html#non-text-contentdef

**Comment by mraccess77** [comment]
Regarding decorative SVGs -- the AAM spec seems to say that certain SVG content is automatically not placed in the accessibility tree if it doesn't meet certain criteria -- therefore it would be already marked as decorative from a defacto standpoint by the browser because of it's absence in the tree.  That's essentially what I'm asking -- is that defactor removal enough or do we need explicit marking.

**Comment by patrickhlauke** [comment]
passing WCAG SCs is results-driven ... so if the end result is that the SVG using content that's automatically NOT placed in the a11y tree is already "hidden" from AT, then that should be a pass, no?

**Comment by mraccess77** [comment]
ACT rule https://act-rules.github.io/rules/e88epe
Pass 4 says this 
Passed Example 4

This svg element that is ignored by assistive technologies because it has no attribute that would give it an accessible name is purely decorative.
